plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/wizards/NewConfigurationFileFormatPage.java

```diff
@@ -0,0 +1,145 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with FeatureIDE.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * See http://featureide.cs.ovgu.de/ for further information.
+ */
+package de.ovgu.featureide.fm.ui.wizards;
+
+import static de.ovgu.featureide.fm.core.localization.StringTable.SELECTED_FILE_ALREADY_EXISTS_;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.eclipse.jface.dialogs.IDialogPage;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+
+import de.ovgu.featureide.fm.core.ExtensionManager.NoSuchExtensionException;
+import de.ovgu.featureide.fm.core.base.impl.ConfigFormatManager;
+import de.ovgu.featureide.fm.core.configuration.XMLConfFormat;
+import de.ovgu.featureide.fm.core.io.IConfigurationFormat;
+
+/**
+ * The NEW wizard page allows setting the container for the new file as well as the file name. The page will only accept file name without the extension OR with
+ * the extension that matches the expected one (.config).
+ * 
+ * @author Christian Becker
+ * @author Jens Meinicke
+ * @author Marlen Bernier
+ * @author Dawid Szczepanski
+ */
+public class NewConfigurationFileFormatPage extends WizardPage {
+
+	private final List<IConfigurationFormat> formatExtensions = ConfigFormatManager.getInstance().getExtensions();
+
+	private Combo formatCombo;
+	
+	String description;
+
+	/**
+	 * Constructor for SampleNewWizardPage.
+	 * 
+	 * @param pageName
+	 */
+	public NewConfigurationFileFormatPage() {
+		super("format");
+		setTitle("Choose Format");
+		setDescription("Select a format for the new configuration file.");
+		
+	}
+
+	/**
+	 * @see IDialogPage#createControl(Composite)
+	 */
+	public void createControl(Composite parent) {
+		GridData gd = new GridData(GridData.FILL_HORIZONTAL);
+		Composite composite = new Composite(parent, SWT.NULL);
+		GridLayout layout = new GridLayout();
+		layout.numColumns = 3;
+		layout.verticalSpacing = 9;
+		composite.setLayout(layout);
+
+		Label label = new Label(composite, SWT.NULL);
+		label.setText("&Format:");
+		formatCombo = new Combo(composite, SWT.BORDER | SWT.SINGLE);
+		formatCombo.setLayoutData(gd);
+		new Label(composite, SWT.NULL);
+
+		initialize();
+		addListeners();
+		dialogChanged();
+		setControl(composite);
+	}
+
+	private void addListeners() {
+		formatCombo.addModifyListener(new ModifyListener() {
+
+			public void modifyText(ModifyEvent e) {				
+				String suffix= getFormat().getSuffix();
+				if( !suffix.equals("xml")) {
```

Comment:
Extremely important: remove the space in front of the exclamation mark

Reply:
Than you very much for pointing this out!